src/Mount.cpp

```diff
@@ -5,10 +5,14 @@
 #include "Mount.hpp"
 #include "Sidereal.hpp"
 
+PUSH_NO_WARNINGS
 #include <AccelStepper.h>
+POP_NO_WARNINGS
 #if (RA_DRIVER_TYPE == DRIVER_TYPE_TMC2209_UART) || (DEC_DRIVER_TYPE == DRIVER_TYPE_TMC2209_UART) || \
   (AZ_DRIVER_TYPE == DRIVER_TYPE_TMC2209_UART) || (ALT_DRIVER_TYPE == DRIVER_TYPE_TMC2209_UART)
+  PUSH_NO_WARNINGS
   #include <TMCStepper.h>   // If you get an error here, download the TMCstepper library from "Tools > Manage Libraries"
+  POP_NO_WARNINGS
 #endif
```

Comment:
in theory we dont need to do thing for each include, we could just wrap all the external lib/framework includes (as block) with this macro

Reply:
Yeah I know, I felt this was a bit cleaner since there's some preprocessor control blocks which break the includes up. I can move them to a single `PUSH`/`POP` if you think that's cleaner

Reply:
I think it would be easier to read and also a clean simple convention: "wrap all external library includes with NO_WARNINGS". Probably an even better solution would be to disable the checks for the external libs per compiler flags in pio ... but I dont know (yet) if it is possible.

Reply:
Yeah I initially tried to have platformio use `-isystem` when including external headers so that GCC suppresses all warnings ([system header documentation](https://gcc.gnu.org/onlinedocs/cpp/System-Headers.html)) but platformio wasn't playing nice... I really delved deep into how platformio works (i.e. [hacking around with SCons](https://docs.platformio.org/en/latest/projectconf/advanced_scripting.html)) but I couldn't get it to do what I wanted. The macros are a pretty ugly solution but I think it's the only maintainable way to do what we want.